daemon/unit_test.c

```diff
@@ -1606,6 +1606,107 @@ int test_sqlite(void) {
     return 1;
 }
 
+int unit_test_bitmap256(void) {
+    fprintf(stderr, "%s() running...\n", __FUNCTION__ );
+
+    BITMAP256 test_bitmap = {0};
+
+    bitmap256_set_bit(&test_bitmap, 0, 1);
+    bitmap256_set_bit(&test_bitmap, 64, 1);
+    bitmap256_set_bit(&test_bitmap, 128, 1);
+    bitmap256_set_bit(&test_bitmap, 192, 1);
+    if (test_bitmap.data[0] == 1)
+        fprintf(stderr, "%s() INDEX 1 is OK\n", __FUNCTION__ );
+    if (test_bitmap.data[1] == 1)
+        fprintf(stderr, "%s() INDEX 65 is OK\n", __FUNCTION__ );
+    if (test_bitmap.data[2] == 1)
+        fprintf(stderr, "%s() INDEX 129 is OK\n", __FUNCTION__ );
+    if (test_bitmap.data[3] == 1)
+        fprintf(stderr, "%s() INDEX 192 is OK\n", __FUNCTION__ );
+
+    uint8_t i=0;
+    do {
+        bitmap256_set_bit(&test_bitmap, i++, 1);
+    } while (i);
```

Comment:
Considering what I am seeing here, we are expecting to create an overflow to stop the loop, can we rewrite the tester for something more soft? :)

Reply:
I agree 100% with your sentence, I was only saying that the way it is written we are forcing an overflow, my suggestion was in the direction to test until the limit `0XFF` instead to force an "error".

Reply:
Added another counter and not rely on uint8_t overflow :smile:

Reply:
Now I will run a final test and approve your PR. 

And for our users that were reading this thread and want to learn a little bit more, I did not approve the PR, because this:

X86_64 computers have different registers, and integer registers are known as `RAX/EAX/AX` and this last can be split into `AH` and `AL`. What I was saying about an overflow that needed to be eliminated, it is because the increasing counting of numbers in AL (uint8_t) until it exceeds the maximum allowed (0xFF) would set a value `AH` and reset completely `AL` to zero, in other words, the loop would stop when `uint8_t` were set to `0` and we started to fill `uint16_t`. 

The last update brought this in a "soft" way. :rofl: